File: app/src/main/java/com/addie/maxfocus/ui/IntroActivity.java
package com.addie.maxfocus.ui;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Build;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import com.addie.maxfocus.R;
import com.github.paolorotolo.appintro.AppIntro;
import com.github.paolorotolo.appintro.AppIntroFragment;
import com.github.paolorotolo.appintro.model.SliderPage;

public class IntroActivity extends AppIntro {
    @Override
    protected void onResume() {
        super.onResume();
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        SliderPage page1 = new SliderPage();
        page1.setTitle(getString(R.string.app_name));
        page1.setDescription("Control your phone usage");
        page1.setBgColor(getResources().getColor(R.color.colorPrimaryDark));
        page1.setImageDrawable(R.drawable.app_icon_large);
        addSlide(AppIntroFragment.newInstance(page1));

        SliderPage pagei = new SliderPage();
        pagei.setTitle("Add app shortcut");
        pagei.setDescription("Add new app shorcuts to the home screen using " + getString(R.string.app_name));
        pagei.setBgColor(getResources().getColor(R.color.colorPrimaryDark));
        pagei.setImageDrawable(R.drawable.replace_icon);

        SliderPage page2 = new SliderPage();
        page2.setTitle("Set a timer");
        page2.setDescription("Before starting an app,specify the time you want to spend on it");
        page2.setBgColor(getResources().getColor(R.color.colorPrimaryDark));
        page2.setImageDrawable(R.drawable.time_dialog_phone);

        SliderPage page3 = new SliderPage();
        page3.setTitle("Get notified");
        page3.setDescription("A dialog will pop up when the specified time has passed");
        page3.setBgColor(getResources().getColor(R.color.colorPrimaryDark));
        page3.setImageDrawable(R.drawable.stop_dialog_new);

        SliderPage page4 = new SliderPage();
        page4.setTitle("All Set!");
        page4.setDescription("Press Done to begin");
        page4.setImageDrawable(R.drawable.ic_check_circle_big_green_128dp);
        page4.setBgColor(getResources().getColor(R.color.colorPrimaryDark));

        // Instead of fragments, you can also use our default slide
        // Just set a title, description, background and image. AppIntro will do the rest
        addSlide(AppIntroFragment.newInstance(pagei));
        addSlide(AppIntroFragment.newInstance(page2));
        addSlide(AppIntroFragment.newInstance(page3));
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            addSlide(PermissionSlide.newInstance(R.layout.fragment_permission_slide));
        }
        addSlide(AppIntroFragment.newInstance(page4));

        // OPTIONAL METHODS
        // Override bar/separator color
//        setBarColor(Color.parseColor("#3F51B5"));
//        setSeparatorColor(Color.parseColor("#2196F3"));
        // Hide Skip/Done button
        showSkipButton(true);
        setProgressButtonEnabled(true);

        showSeparator(false);

        // SHOW or HIDE the statusbar

        showStatusBar(false);
        // Animations -- use only one of the below. Using both could cause errors.
        setFadeAnimation(); // OR
//        setZoomAnimation(); // OR
//        setFlowAnimation(); // OR
//        setSlideOverAnimation(); // OR
//        setDepthAnimation(); // OR
//        setCustomTransformer(yourCustomTransformer);

        // Permissions -- takes a permission and slide number
//        askForPermissions(new String[]{
//                Manifest.permission.CAMERA}, 1);
    }

    @Override
    public void onSkipPressed(Fragment currentFragment) {
        super.onSkipPressed(currentFragment);
        //TODO : Change to last slide
        getPager().setCurrentItem(5);

    }

    @Override
    public void onDonePressed(Fragment currentFragment) {
        super.onDonePressed(currentFragment);
        startActivity(new Intent(this, MainActivity.class));
        finish();
        final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
        prefs.edit().putBoolean(getString(R.string.tutorial_seen_key),true).apply();

    }

    @Override
    public void onSlideChanged(@Nullable Fragment oldFragment, @Nullable Fragment newFragment) {
        super.onSlideChanged(oldFragment, newFragment);
    }

    @Override
    public void onNextPressed() {
        super.onNextPressed();
    }

}

File: app/src/main/java/com/addie/timesapp2/ui/IntroActivity.kt
/*
 * MIT License
 *
 * Copyright (c) 2018 aSoft
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.addie.timesapp2.ui

import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.preference.PreferenceManager
import androidx.fragment.app.Fragment
import com.addie.timesapp2.R
import com.github.paolorotolo.appintro.AppIntro
import com.github.paolorotolo.appintro.AppIntroFragment
import com.github.paolorotolo.appintro.model.SliderPage

/**
 * An introductory activity that explains the method of use of the app
 */
class IntroActivity : AppIntro() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mCallingClass = ""
        if (intent.hasExtra(CALLING_CLASS_KEY)) {
            mCallingClass = intent.getStringExtra(CALLING_CLASS_KEY)
            if (mCallingClass == "SettingsFragment") {
                val preferences = androidx.preference.PreferenceManager.getDefaultSharedPreferences(this@IntroActivity)
                preferences.edit().putBoolean(getString(R.string.pref_display_tap_target_apps), true).apply()

                preferences.edit().putBoolean(getString(R.string.pref_display_tap_target_time_dialog), true).apply()
            }
        }

        val page1 = SliderPage()
        page1.title = getString(R.string.app_name)
        page1.description = getString(R.string.intro_page1_desc)
        page1.bgColor = resources.getColor(R.color.colorPrimaryDark)
        page1.imageDrawable = R.drawable.app_icon_large

        val page2 = SliderPage()
        page2.title = getString(R.string.intro_page2_title)
        page2.description = getString(R.string.intro_page2_desc)
        page2.bgColor = resources.getColor(R.color.colorPrimaryDark)
        page2.imageDrawable = R.drawable.replace_icon

        val page3 = SliderPage()
        page3.title = getString(R.string.intro_page3_title)
        page3.description = getString(R.string.intro_page3_desc)
        page3.bgColor = resources.getColor(R.color.colorPrimaryDark)
        page3.imageDrawable = R.drawable.time_dialog_phone

        val page4 = SliderPage()
        page4.title = getString(R.string.intro_page4_title)
        page4.description = getString(R.string.intro_page4_desc)
        page4.bgColor = resources.getColor(R.color.colorPrimaryDark)
        page4.imageDrawable = R.drawable.stop_dialog

        val page5 = SliderPage()
        page5.title = getString(R.string.intro_page5_title)
        page5.description = getString(R.string.intro_page5_desc)
        page5.imageDrawable = R.drawable.ic_check_circle_big_green_128dp
        page5.bgColor = resources.getColor(R.color.colorPrimaryDark)

        addSlide(AppIntroFragment.newInstance(page1))
        addSlide(AppIntroFragment.newInstance(page2))
        addSlide(AppIntroFragment.newInstance(page3))
        addSlide(AppIntroFragment.newInstance(page4))
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            addSlide(UsagePermissionSlide.newInstance(R.layout.fragment_permission_slide))
        }
//        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.)
        addSlide(AppIntroFragment.newInstance(page5))

        showSkipButton(true)
        isProgressButtonEnabled = true
        showSeparator(false)
        showStatusBar(false)
        setFadeAnimation()

    }

    override fun onSkipPressed(currentFragment: Fragment?) {
        super.onSkipPressed(currentFragment)
        getPager().currentItem = 4

    }

    override fun onDonePressed(currentFragment: Fragment?) {
        super.onDonePressed(currentFragment)

        val prefs = PreferenceManager.getDefaultSharedPreferences(this)
        prefs.edit().putBoolean(getString(R.string.tutorial_seen_key), true).apply()

        val mainIntent = Intent(this, MainActivity::class.java)

        // Display tutorials preference has been selected from preference screen
        if (mCallingClass == "SettingsFragment") {
            // Finish all activities
            finishAffinity()
            startActivity(mainIntent)
        } else {
            startActivity(mainIntent)
            finish()
        }//Default behavior

    }

    companion object {
        private val CALLING_CLASS_KEY = "calling_class"
        private var mCallingClass: String? = null
    }

}

